internal/pkg/agent/install/install_windows.go

```diff
@@ -21,6 +22,13 @@ import (
 	"github.com/elastic/elastic-agent/version"
 )
 
+const (
+	// Conforming to https://learn.microsoft.com/en-us/troubleshoot/windows-server/active-directory/naming-conventions-for-computer-domain-site-ou#domain-names
+	// domain names can contain all alphanumeric characters except for the extended characters that appear in the Disallowed characters list. Names can contain a period, but names can't start with a period.
+	// Disallowed characters: [, ~ : @ # $ % ^ ' . ( ) { } _ {whitespace} \ / ]
+	activeDirectoryUsername = `^[A-Za-z0-9]+(?:\.[A-Za-z0-9]+)*\\[A-Za-z0-9.-]{1,104}$`
```

Comment:
Why do we need to validate this instead of letting Windows tell us the characters are invalid when we try to install?

Reply:
because I find it good practice to validate on input level rather than let system find failures for you

Reply:
Yes, with the trade off that we have to encode the rules of the system into our code, and the two things can get out of sync. Probably this one is OK, but if the error message from the system were reasonably clear, this is some code we wouldn't need to maintain.